pkg/service/service.go

```diff
@@ -1120,59 +1029,79 @@ func UpdateServicesWithOwnerReferences(client *kclient.Client, k8sComponents []d
 		if found {
 			continue
 		}
-		u.SetOwnerReferences(append(u.GetOwnerReferences(), ownerReference))
+		d.SetOwnerReferences(append(d.GetOwnerReferences(), ownerReference))
 
-		err = client.UpdateDynamicResource(group, version, resource, crdName, u)
+		err = client.UpdateDynamicResource(gvr.Resource.Group, gvr.Resource.Version, gvr.Resource.Resource, u.GetName(), d)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func getCRDName(crd map[string]interface{}) (string, bool) {
-	metadata, ok := crd["metadata"].(map[string]interface{})
-	if !ok {
-		return "", false
+func isLinkResource(kind string) bool {
+	return kind == "ServiceBinding"
+}
+
+// GetGVKFromUnstructured takes an unstructured struct and returns group, version and kind information as strings
+// It assumes group to be "core" for cases where metadata.apiVersion can't be split using forward slash, e.g. 'v1'
+func GetGVKFromUnstructured(u unstructured.Unstructured) schema.GroupVersionKind {
```

Comment:
this version already exists in https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured#Unstructured.GroupVersionKind

Reply:
https://github.com/openshift/odo/pull/4964/commits/8270bb639dbd38589082f73b590955c489cdddd7